### ISSUE TITLE
multiple dom0 attach failed with SMAPIv3

### ISSUE THREAD

**Issue by gthvn1** [xapi-project/xen-api#5521]
### tl;dr:

When comparing an old version of Xen API (v1.249.32) and the new version (master) I see that when I tried to plug a second VBD a message `VDI.activate3` is sent in the message-switch with new version and I got an error `VDIINUSE`. It is not the case with old Xen API. The test is to reproduce multiple dom0 attachment of the quicktest.

---

  Here is the scenario (this is the scenario of the "multiple dom0 attach quicktest") 
- I created a "zfs-ng" storage that is smapiv3 (but the issue is in datapath so I guess it is the same with another SR type)
- I Created one VDI on this SR
- Then I Create two VBDs  with the same vm uuid and vdi uuid and plug them
    - On xcp-ng 8.2 (Xen API tag v1.249.32) it works fine
    - On xcp-ng 8.3 (~master) I have an error `VDIINUSE`

  When I plug the first VBD, `VDI.activate3` message is sent on message-switch in both cases (old and new version of Xen API). But when I plugged the second VBD,  in the case if the new Xen API (master) a `VDI.activate3` message is also sent on message-switch.  It is not the case for the older Xen API (1.249). So with the new Xen API the datapath active is called again. The problem is that when the SMAPIv3 datapath it tries to activate the VDI it first checks if VDI is already `active on` a host and failed if it is the case. Thus I have an error `VDIINUSE`. 

  As said previously it works on the older version of Xen API because at some point the Xen API doesn't push any message into the message switch to activate the VDI during the second plug. I tried to check in the code the difference between `storage_mux.ml:VDI.activate3` (the code of the master) and `storage_impl:VDI.activate3` in the former Xen API. But I didn't find why the message is send in one case and not in another...

  My question is: Is it normal to see a message of activation when the second VBD plug is done with the new Xen API? Any idea why the older version of Xen API doesn't send this activation message when we plugged the second VBD and why it is not the case with the master?

Any enlightenment is welcome :)
Cheers,
Guillaume

**Comment by gthvn1** [comment]
Ok in fact after reading the SMAPIv3 twice and specifically the method activate the function should be idempotent. So the issue is in SMAPIv3. So I close the issue :)
